tests/unit/s2n_x509_validator_test.c

```diff
@@ -1609,6 +1730,114 @@ int main(int argc, char **argv)
         s2n_x509_trust_store_wipe(&trust_store);
     };
 
+    /**
+     * Test OCSP validation at various offsets from update times
+     * libcrypto ASN1 comparison calculates differences in terms of days and seconds,
+     * so try mocking the system time to a collection of more than day & less
+     * day differences.
+     * The T's in the below diagram represent test cases that should fail
+     * The F's represent test cases that should succeed
+     *    S2N_ERR_CERT_INVALID                            S2N_ERR_CERT_EXPIRED
+     *          |   |                                        |   |
+     *          v   v                                        v   v
+     *          F   F   T   T                        T   T   F   F
+     *          v   v   v   v                        v   v   v   v
+     * <----------|---|---|----------------------------|---|---|--->
+     *                ^                                    ^
+     *       this update                                 next update
+     *                |---|
+     *                  one day
+     *
+     * If this test is failing make sure that the this_update_timestamp_nanoseconds
+     * matches the actual timestamp of ocsp_response_early_expire.der
+     *
+     * openssl ocsp -respin ocsp_response_early_expire.der -text -noverify | grep "This Update"
+     */
+    {
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t this_update_timestamp_nanoseconds = (uint64_t) 1682719916 * ONE_SEC_IN_NANOS;
+
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t next_update_timestamp_nanoseconds = (uint64_t) 2082838316 * ONE_SEC_IN_NANOS;
+
+        uint64_t one_hour_nanoseconds = (uint64_t) 60 * 60 * ONE_SEC_IN_NANOS;
+        uint64_t one_day_nanoseconds = 24 * one_hour_nanoseconds;
+
+        size_t test_cases = 8;
```

Comment:
nit: This seems like it could be a little risky, since someone could add a test and forget to increase `test_cases`. Could the length be checked on one of the arrays with `s2n_array_len` instead?